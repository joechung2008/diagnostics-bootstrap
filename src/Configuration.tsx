import { Table } from "react-bootstrap";

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const items = Object.entries(config).reduce<KeyValuePair<string>[]>(
    (previous, [key, value]) => [...previous, { key, value }],
    []
  );

  return (
    <div>
      <h3>Configuration</h3>
      <Table>
        <thead>
          <tr>
            <th>Key</th>
            <th>Value</th>
          </tr>
        </thead>
        <tbody>
          {items.map((item, idx) => (
            <tr key={idx}>
              <td>{item.key}</td>
              <td>{item.value}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default Configuration;
