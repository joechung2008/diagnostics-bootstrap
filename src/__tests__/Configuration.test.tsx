import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Configuration from "../Configuration";

describe("Configuration", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Configuration config={{ key1: "value1", key2: "value2" }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
