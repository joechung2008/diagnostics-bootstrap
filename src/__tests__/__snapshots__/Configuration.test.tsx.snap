// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Configuration > renders correctly 1`] = `
<div>
  <h3>
    Configuration
  </h3>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Key
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          key1
        </td>
        <td>
          value1
        </td>
      </tr>
      <tr>
        <td>
          key2
        </td>
        <td>
          value2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
